samtranslator/model/sam_resources.py

```diff
@@ -1276,28 +1275,18 @@ def to_cloudformation(self, **kwargs):  # type: ignore[no-untyped-def]
             always_deploy=self.AlwaysDeploy,
         )
 
-        (
-            rest_api,
-            deployment,
-            stage,
-            permissions,
-            domain,
-            basepath_mapping,
-            route53,
-            usage_plan_resources,
-        ) = api_generator.to_cloudformation(redeploy_restapi_parameters, route53_record_set_groups)
+        generated_api_resources = api_generator.to_cloudformation(
+            redeploy_restapi_parameters, route53_record_set_groups
+        )
 
-        resources.extend([rest_api, deployment, stage])
-        resources.extend(permissions)
-        if domain:
-            resources.extend([domain])
-        if basepath_mapping:
-            resources.extend(basepath_mapping)
-        if route53:
-            resources.extend([route53])
-        # contains usage plan, api key and usageplan key resources
-        if usage_plan_resources:
-            resources.extend(usage_plan_resources)
+        resources: List[Any] = []
+
+        for resource in generated_api_resources:
+            if resource:
+                if isinstance(resource, (list, tuple)):
```

Comment:
Why does `to_cloudformation` return both collections and individual `Resource`s?

Reply:
See my above comment. The tuple it was using before includes both collections and individual resources. Will refactor that.